packages/headless/src/features/insight-search/insight-search-request.test.ts

```diff
@@ -397,4 +398,32 @@ describe('insight search request', () => {
       expect(params.context).toBe(expectedContext);
     });
   });
+
+  it('should enable #queryCorrection if did you mean is enabled and #queryCorrectionMode is `next`', async () => {
+    state.didYouMean.enableDidYouMean = true;
+    state.didYouMean.queryCorrectionMode = 'next';
+    expect(
+      (await buildInsightBaseRequest(state)).request.queryCorrection?.enabled
+    ).toBe(true);
+  });
+
+  it('should enable #automaticallyCorrect if did you mean is enabled and #queryCorrectionMode is `next` and #automaticallyCorrectQuery is true', async () => {
```

Comment:
In the function to build the request, the value of `automaticallyCorrect ` is set regardless of if #queryCorrectionMode is `next` or `legacy` so there is either something to fix/improve in the test of the function.

Reply:
im following the same pattern of tests as in the `search` requests.

It testing this block: 
<img width="546" alt="image" src="https://github.com/user-attachments/assets/87a4e40c-5e65-45ca-8a7a-90b0e3ec040c">